Sources/MapboxNavigation/PreviewViewController.swift

```diff
@@ -0,0 +1,633 @@
+            return
+        }
+        
+        // Update user puck to the most recent location.
+        navigationView.navigationMapView.moveUserLocation(to: location, animated: true)
+        
+        speedLimitView.signStandard = notification.userInfo?[PassiveLocationManager.NotificationUserInfoKey.signStandardKey] as? SignStandard
+        speedLimitView.speedLimit = notification.userInfo?[PassiveLocationManager.NotificationUserInfoKey.speedLimitKey] as? Measurement<UnitSpeed>
+        
+        let roadNameFromStatus = notification.userInfo?[PassiveLocationManager.NotificationUserInfoKey.roadNameKey] as? String
+        if let roadName = roadNameFromStatus?.nonEmptyString {
+            let representation = notification.userInfo?[PassiveLocationManager.NotificationUserInfoKey.routeShieldRepresentationKey] as? VisualInstruction.Component.ImageRepresentation
+            wayNameView.label.updateRoad(roadName: roadName, representation: representation)
+            wayNameView.containerView.isHidden = false
+        } else {
+            wayNameView.text = nil
+            wayNameView.containerView.isHidden = true
+        }
+        
+        updateViewportDataSource(for: location)
+    }
+    
+    @objc func orientationDidChange(_ notification: Notification) {
+        let passiveLocationProvider = navigationView.navigationMapView.mapView.location.locationProvider as? PassiveLocationProvider
+        updateViewportDataSource(for: passiveLocationProvider?.locationManager.location)
+        
+        // In case if routes are already shown and orientation changes - fit camera so that all
+        // routes fit into available space.
+        if case .routesPreviewing(let previewOptions) = state {
+            fitCamera(to: previewOptions.routeResponse)
+        }
+    }
+    
+    func updateViewportDataSource(for location: CLLocation?) {
+        guard let location = location,
+              let navigationViewportDataSource = navigationView.navigationMapView.navigationCamera.viewportDataSource as? NavigationViewportDataSource else { return }
+        
+        let topInset: CGFloat = UIScreen.main.bounds.height - view.safeAreaInsets.bottom - 150.0
+        let bottomInset: CGFloat = 149.0 + view.safeAreaInsets.bottom
+        let leftInset: CGFloat = 50.0
+        let rightInset: CGFloat = 50.0
+        
+        navigationViewportDataSource.followingMobileCamera.padding = UIEdgeInsets(top: topInset,
+                                                                                  left: leftInset,
+                                                                                  bottom: bottomInset,
+                                                                                  right: rightInset)
+        
+        navigationViewportDataSource.followingMobileCamera.bearing = location.course
+        navigationViewportDataSource.followingMobileCamera.pitch = 40.0
```

Comment:
Because when we receive the location updates in passive mode, it may still in browsing mode, may still need to be `0.0`. for following and previewing mode, it could be `40.0` .